test/extensions/filters/http/adaptive_concurrency/controller/gradient_controller_test.cc

```diff
@@ -107,6 +107,11 @@ class GradientControllerTest : public testing::Test {
             .value());
   }
 
+  template <typename DurationType> void advanceTimeAndLoop(DurationType duration) {
+    time_system_.advanceTimeAsync(duration);
```

Comment:
WDYT about having `advanceTimeAsync` take a dispatcher and whether to run it blocking or non-blocking? It seems like it would be almost always broken to not do this pattern when using this API? If there are some cases I guess the dispatcher could be optional in the API but it would at least make test writers think about it?

Reply:
@jmarantz @antoniovicente ping on this one. I see a new commit but I'm unsure of the plan here. If this is not needed we can merge but wanted to check.

/wait-any

Reply:
Sorry, was doing some research in order to provide a complete reply to your comment.

As you noticed I rolled back this file since the changes to it are no longer required once I moved to updating monotonic time directly instead of updating next_monotonic_time and propagating it to event loops next time they run.

The idea of taking a dispatcher on advanceTimeAsync is an interesting suggestion.  I found that most uses of this function happen in cases where there is a single active Dispatcher, but there's 1 exception to this in ServerStatsTest.FlushStats.  Still it makes sense to have a SimulatedTimeSystem function that moves time forward and then runs the dispatcher event loop.  Preferences for doing that change in this PR or a followup cleanup?

Reply:
> Preferences for doing that change in this PR or a followup cleanup? 

Up to you, I'm fine either way. Thank you!

Reply:
I'll do the cleanup as a followup.